jekyll/_cci2/bundling-config.md

```diff
@@ -0,0 +1,905 @@
+  s3sync:
+    parameters:
+      from:
+        type: string
+      to:
+        type: string
+      overwrite:
+        default: false
+        type: boolean
+    steps:
+      - run:
+          name: Deploy to S3
+          command: "aws s3 sync << parameters.from >> << parameters.to >><<# parameters.overwrite >> --delete<</ parameters.overwrite >>"
+executors:
+  aws:
+    docker:
+      - image: cibuilds/aws:1.15
+jobs:
+  deploy2s3:
+    executor: aws
+    steps:
+      - s3sync:
+          from: .
+          to: "s3://mybucket_uri"
+          overwrite: true
+```
+
+## Using the `parameters` Declaration
+
+**Note:** The `parameters` declaration is available in configuration version 2.1 and later.
+
+Many config elements can be authored to be invocable in your `config.yml` file with specified parameters. Parameters are declared by name as the keys in a map that are all immediate children of the `parameters` key under a job, command, or executor. 
+
+For instance, the following shows declaring a parameter `foo` in the definition of a command `bar`:
+
+```yaml
+commands:
+  bar:
+    parameters:
+      foo:
+        description: this parameter is used for a thing
+        default: "I'm a default value"
+        type: string
+
+    steps:
+      - run: echo '<< parameters.foo >>'
+```
+
+In the above example a value for the parameter `foo` can be passed when invoking the command. For instance, passing the parameter `foo` would look something like:
+
+```yaml
+jobs:
+  myjob:
+    docker:
+      - image: circleci/ruby:2.4-node
+    steps:
+      - bar:
+          foo: "I'm the passed value"
+```
+
+### Parameter Syntax
+{:.no_toc}
+
+A parameter can have the following keys as immediate children:
+
+| Key Name    | Description                                                                                   | Default value |
+|-------------|-----------------------------------------------------------------------------------------------|---------------|
+| description | Optional. Used to generate documentation for your orb.                                        | N/A           |
+| type        | Required. Currently "string", "boolean", "enum", and "steps" are supported                           | N/A           |
+| default     | The default value for the parameter. If not present, the parameter is implied to be required. | N/A           |
+
+### Parameter Types
+{:.no_toc}
+
+This section describes the types of parameters and their usage.
+
+#### String
+{:.no_toc}
+
+Basic string parameters are described below:
+
+```yaml
+commands:
+  copy-markdown:
+    parameters:
+      destination:
+        description: destination directory
+        type: string
+        default: docs
+    steps:
+      - cp *.md << destination >>
+```
+
+Strings should be quoted if they would otherwise represent another type (such as boolean or number) or if they contain characters that have special meaning in yaml. In all other instances, quotes are optional.
+
```

Comment:
We should probably add here that "Empty strings are treated as a falsy value in evaluation of `when` clauses, and all other strings are treated as truthy"

Reply:
thanks, added